server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -656,6 +658,26 @@ public IndexSearcher wrap(IndexSearcher searcher) throws EngineException {
         IOUtils.close(store, engine);
     }
 
+
+    public void testTranslogRecoveryFailure() throws Exception {
```

Comment:
Although this test makes sense, the bulk of this change is to the `Translog` so I would expect there to be changes/additions to `TranslogTests`.

Reply:
@DaveCTurner I'll make sure that each test that deals with TranslogCorruptedException will check that it contains correct source in it. Adding more tests for scenarios when TranslogCorruptedException could occur seems to much for this PR, because I don't introduce new throw statements, but instead just add some information to TranslogCorruptedException